src/model/clients/FetchAdapter.ts

```diff
@@ -0,0 +1,79 @@
+import { GenericResponse } from 'model/clients/GenericResponse';
+import { HttpClient } from 'model/clients/HttpClient';
+
+/**
+ * This is an adapter for using an HttpClient with the same
+ * interface as the global `fetch` function.
+ */
+class FetchAdapter<T extends HttpClient = HttpClient> {
```

Comment:
This makes us able to use our classes based on the `HttpClient` base class with the CP Client, as a replacement for the global `fetch` method.